IPAM pools followups

tklauser opened the pull request:
Address out-of-band review comments made on #22762. See commits for details.

For #25470

tklauser committed:
ipam/service/allocator: remove incorrect comments

These interface methods are not only used for testing. Remove them.

Suggested-by: Sebastian Wicki <sebastian@isovalent.com>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
ipam/cidrset: use (*CidrSet).getIndexForIP directly

Instead of constructing a *net.IPnet whose .Mask is never used call
(*CidrSet.getIndexForIP directly.

This corresponds to upstream PR
https://github.com/kubernetes/kubernetes/pull/117931

Signed-off-by: Tobias Klauser <tobias@cilium.io>

gandro commented on pkg/ipam/multipool.go:
```diff
@@ -453,20 +453,19 @@ func (m *multiPoolManager) allocateIP(ip net.IP, owner string, poolName Pool, fa
 	pool := m.poolByFamilyLocked(poolName.String(), family)
 	if pool == nil {
-		return fmt.Errorf("unable to release IP %s of unknown pool %q (family %s)", ip, poolName, family)
+		return
```
I think we should at least emit a warning here. This is something that indicates desynced state after all. How could we have released a pool if an owner still thinks it has an IP allocated from it?

tklauser replied:
Good point, added a warning.

gandro reviewed (commented):
Thanks!

aojea commented on pkg/ipam/service/allocator/interfaces.go:
```diff
@@ -11,11 +11,7 @@ type Interface interface {
 	Release(int) error
 	ForEach(func(int))
-
-	// For testing
```
can you send the same change in kubernetes/kubernetes?

tklauser replied:
Opened https://github.com/kubernetes/kubernetes/pull/118171

tklauser committed:
ipam: remove Interface.Release error return value

The release operation won't actually fail and having the error return
leads to awkward errors handling. The underlying ipallocator.Range and
allocator.AllocatorBitmap implementations always return nil errors
already, so there is no point in returning an error.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
